dra-evolution/pkg/api/capacity_types.go

```diff
@@ -3,168 +3,134 @@ package api
+	//
+	// This is currently required, but this might get relaxed in the future.
+	NodeName string `json:"nodeName"`
 
-	// Attributes contains device attributes that are common to all devices
-	// in the pool.
-	// +optional
-	// +listType=atomic
-	Attributes []Attribute `json:"attributes,omitempty"`
+	// POTENTIAL FUTURE EXTENSION: NodeSelector *v1.NodeSelector
 
-	// SharedResources are pooled resources that are shared by all devices
-	// in the pool. This is typically used when representing a
-	// partitionable device, and need not be populated otherwise.
+	// DriverName identifies the DRA driver providing the capacity information.
+	// A field selector can be used to list only ResourcePool
+	// objects with a certain driver name.
 	//
-	// +optional
-	// +listType=atomic
-	SharedResources []ResourceCapacity `json:"sharedResources,omitempty"`
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	DriverName string `json:"driverName" protobuf:"bytes,3,name=driverName"`
 
-	// Devices contains the individual devices in the pool.
+	// Devices lists all available devices in this pool.
 	//
-	// +required
-	// +listType=atomic
+	// Must not have more than 128 entries.
 	Devices []Device `json:"devices,omitempty"`
-}
 
-// DevicePoolStatus contains the state of the pool as last reported by the
-// driver. Note that this will not include the allocations that have been made
-// by the scheduler but not yet seen by the driver. Thus, it is NOT sufficient
-// to make future scheduling decisions.
-type DevicePoolStatus struct {
-	DeviceStatuses []AllocatedDevice `json:"deviceStatuses,omitempty"`
+	// FUTURE EXTENSION: some other kind of list, should we ever need it.
+	// Old clients seeing an empty Devices field can safely ignore the (to
+	// them) empty pool.
 }
 
-// AllocatedDevice represents a device that has been allocated from the pool.
-type AllocatedDevice struct {
-	Name string
-	// Conditions contains the latest observation of the device's state.
-	Conditions []metav1.Condition `json:"conditions"`
+const ResourcePoolMaxDevices = 128
 
-	// ClaimUIDs contains the UIDs of the claims to which this device
-	// is allocated.
-	ClaimUIDs []types.UID
-}
-
-// Device is used to track individual devices in a pool.
+// Device represents one individual hardware instance that can be selected based
+// on its attributes.
 type Device struct {
-	// Name is a driver-specific identifier for the device.
-	// +required
-	Name string `json:"name"`
+	// Name is unique identifier among all devices managed by
+	// the driver on the node. It must be a DNS label.
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
 
-	// Attributes contain additional metadata that can be used in
-	// constraints. If an attribute name overlaps with the pool attribute,
-	// the device attribute takes precedence.
+	// Attributes defines the attributes of this device.
+	// The name of each attribute must be unique.
```

Comment:
Did we decide NOT to bring back per-pool attribs?

Reply:
No final decision. It's just not in the current prototype. Keep it simple? :grin:

Reply:
We can always add later, I am resolving this comment.